sktime/series_as_features/compose/_pipeline.py

```diff
@@ -35,55 +37,126 @@ class FeatureUnion(_FeatureUnion, _PanelToPanelTransformer):
-        super().fit(X, y, **fit_params)
-        self._is_fitted = True
+        transformer_list = self.transformer_list
+
+        # clone and fit transformers in transformer_list, store fitted copies
+        transformer_list_ = [clone(trafo[1]).fit(X, y) for trafo in transformer_list]
+        self.transformer_list_ = transformer_list_
+
         return self
 
-    def transform(self, X):
+    def _transform(self, X, y):
         """Transform X separately by each transformer, concatenate results."""
-        self.check_is_fitted()
-        return super().transform(X)
+        # retrieve fitted transformers, apply to the new data individually
+        transformer_list_ = self.transformer_list_
+        Xt_list = [trafo.transform(X, y) for trafo in transformer_list_]
 
-    def fit_transform(self, X, y, **fit_params):
-        """Transform X separately by each transformer, concatenate results."""
-        return self.fit(X, y, **fit_params).transform(X)
+        transformer_names = [x[0] for x in self.transformer_list]
 
-    def _hstack(self, Xs):
-        """
-        Stacks X horizontally.
+        Xt = pd.concat(
+            Xt_list, axis=1, keys=transformer_names, names=["transformer", "variable"]
+        )
+
+        if self.flatten_transform_index:
+            flat_index = pd.Index("__".join(x) for x in Xt.columns)
+            Xt.columns = flat_index
 
-        Supports input types (X): list of
-            numpy arrays, sparse arrays and DataFrames.
+        return Xt
+
+
+    def get_params(self, deep=True):
+        """Get parameters of estimator in `_forecasters`.
+
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained sub-objects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
         """
-        if any(sparse.issparse(f) for f in Xs):
-            Xs = sparse.hstack(Xs).tocsr()
+        return self._get_params("transformer_list", deep=deep)
 
-        types = {type(X) for X in Xs}
-        if self.preserve_dataframe and (pd.Series in types or pd.DataFrame in types):
-            return pd.concat(Xs, axis=1)
+    def set_params(self, **kwargs):
+        """Set the parameters of estimator in `_forecasters`.
+
+        Valid parameter keys can be listed with ``get_params()``.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self._set_params("transformer_list", **kwargs)
+        return self
 
-        else:
-            return np.hstack(Xs)
+    @classmethod
+    def get_test_params(cls):
+        """Test parameters for FeatureUnion."""
+        from sklearn.preprocessing import StandardScaler
+
+        SERIES_TO_SERIES_TRANSFORMER = StandardScaler()
+
+        from sktime.transformations.panel.compose import SeriesToSeriesRowTransformer
```

Comment:
Move to top :)

Reply:
Per design, imports for the test instances should be inside the testing methods.
Perhaps I should clarify this in the extension template?

Reply:
I made a PR that clarifies this in the extension template:
https://github.com/alan-turing-institute/sktime/pull/1811